research/flamby/fed_isic2019/fedavg/server.py

```diff
@@ -36,6 +37,28 @@ def main(config: Dict[str, Any], server_address: str, checkpoint_stub: str, run_
     client_manager = SimpleClientManager()
     model = Baseline()
 
+    log(INFO, f"if pretrain: {pretrain}")
+    if pretrain:
+        dir = (
+            "/ssd003/projects/aieng/public/FL_env/models/fed_isic2019/fedavg/hp_sweep_results/lr_0.001/"
+            + run_name
+            + "/server_best_model.pkl"
+        )
+        fedavg_model_state = torch.load(dir).state_dict()
+        model_state = model.state_dict()
+        matching_state = {}
+        for k, v in fedavg_model_state.items():
+            if k in model_state:
+                if v.size() == model_state[k].size():
+                    matching_state[k] = v
+                elif model_state[k].size()[1:] == v.size()[1:]:
+                    repeat = model_state[k].size()[0] // v.size()[0]
+                    original_size = tuple([1] * (len(model_state[k].size()) - 1))
+                    matching_state[k] = v.repeat((repeat,) + original_size)
```

Comment:
I'm a little confused as to what is happening here? A comment or set of comments would be useful to help illustrate what these calculations are doing.

Reply:
Again I will delete them for now. But for clarification, basic idea was because we were loading fedavg and model was not fenda model I needed to do such a checking and duplications in weights to not get error in loading from state dict.